Lib/test/test_tempfile.py

```diff
@@ -427,6 +450,14 @@ def test_choose_directory(self):
         finally:
             os.rmdir(dir)
 
+    def test_choose_pathlike_directory(self):
+        # _mkstemp_inner can create files in a user-selected pathlike directory
+        dir = pathlib.Path(tempfile.mkdtemp())
+        try:
```

Comment:
Instead of a try/finally, use [`unittest.TestCase.addCleanup()`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.addCleanup) (basically you shouldn't have to use any try/finally blocks in your test code thanks to that method).